### PR TITLE
Update docs for 21.08 release

### PR DESCRIPTION
**Pull request by sameerz**
Signed-off-by: Sameer Raheja <sraheja@nvidia.com>

Documentation updates for 21.08, prior to release.

### CODE REVIEW

**Review by jlowe — COMMENTED** [review]
The changes seem OK to me, but I'm wondering if we should be making these changes at the same time we change the pom.xml to reflect what these docs are saying.  We've already been bitten in the past with users coming to the dev branch page in github before the release is ready and getting confused.  Plus if one actually builds on `branch-21.08` in the future, they will produce a `21.08.0-SNAPSHOT` artifact, not a `21.08.0` version.

So I'm wondering if these fixups should appear as a separate commit in the merge to the main branch when we're ready to release, just as the pom changes to remove SNAPSHOT also occur.

Note that it would be OK to remove the cudf SNAPSHOT text as soon as branch-21.08 moves to the official cudf-21.08.0 artifact.  All I'm advocating for here is to tie the doc changes to the pom.xml changes that are corresponding.  Disconnecting them can lead to confusion, as we've encountered in the past.

**Comment by sameerz** [comment]
> The changes seem OK to me, but I'm wondering if we should be making these changes at the same time we change the pom.xml to reflect what these docs are saying. We've already been bitten in the past with users coming to the dev branch page in github before the release is ready and getting confused. Plus if one actually builds on branch-21.08 in the future, they will produce a 21.08.0-SNAPSHOT artifact, not a 21.08.0 version.
> 
> So I'm wondering if these fixups should appear as a separate commit in the merge to the main branch when we're ready to release, just as the pom changes to remove SNAPSHOT also occur.

The question in the past, as I understood it, was whether to make documentation changes after we cut a drop to QA, or giving QA the changes as close as possible to the final version to test.  This PR errs on the side of giving QA the doc changes, while recognizing there is a gap period where the links do not match what might be available.  It also gives us a chance to "test" the documentation during the virtual review next week.  I am open to changing our approach.

**Review by jlowe — APPROVED** [review]
We can proceed as usual for now, although I'd like to see this be in draft for now and be one of the last commits.  The pom hasn't even changed to pull in cudf 21.08.0, for example, so it's a bit early to commit this, IMO.

**Comment by sameerz** [comment]
build

File: docs/download.md [hunk 1]
```diff
@@ -18,6 +18,61 @@ cuDF jar, that is either preinstalled in the Spark classpath on all nodes or sub
...
+  * Additional filtering capabilities
+  * Reporting on data types
+  * Reporting on read data formats
+  * Abillity to run the qualification tool on Spark 2.x logs
+  * Ability to run the tool on Apache Spark 3.x, AWS EMR 6.3.0, Dataproc 2.0, Microsoft Azure, and
+    Databricks 7.3 and 8.2 logs
+  * Improved qualification tool performance
+
+For a detailed list of changes, please refer to the
+[CHANGELOG](https://github.com/NVIDIA/spark-rapids/blob/main/CHANGELOG.md).
```
Comment (Salonijain27) [inline docs/download.md:74]: Will the Changelog.md file be updated before this PR gets merged?

Comment (sameerz) [reply docs/download.md:74]: The changelog PR #3083 is in the process of getting reviewed and will be merged soon.  I expect we might do a few more changelog updates before we are done.

**Review by Salonijain27 — APPROVED** [review]
lgtm

File: docs/download.md [hunk 2]
```diff
@@ -18,6 +18,61 @@ cuDF jar, that is either preinstalled in the Spark classpath on all nodes or sub
...
+	Apache Spark 3.0.1, 3.0.2, 3.0.3, 3.1.1, 3.1.2, Cloudera CDP 7.1.6, 7.1.7, Databricks 7.3 ML LTS or 8.2 ML Runtime, and GCP Dataproc 2.0
+
+	Apache Hadoop 2.10+ or 3.1.1+ (3.1.1 for nvidia-docker version 2)
+
+	Python 3.6+, Scala 2.12, Java 8
+
+*Some hardware may have a minimum driver version greater than v450.80.02+.  Check the GPU spec sheet
+for your hardware's minimum driver version.
+
+### Download v21.08.00
```
Comment (viadea) [inline docs/download.md:43]:
```suggestion
### Download v21.08.0
```

**Comment by sameerz** [comment]
build

**Comment by pxLi** [comment]
cudf and cudfjni 21.08.0 has not be released yet (plan on Aug 5), we will update pom dependency after cudfjni 21.08.0 available in maven central. I agree that we can keep the development branch to stick w/ `-SNAPSHOT`, and only update doc to release branch (main)

**Review by jlowe — DISMISSED** [review]

**Comment by pxLi** [comment]
hi looks like this PR did not include version update to https://github.com/NVIDIA/spark-rapids/blob/branch-21.08/docs/get-started/getting-started-on-prem.md#download-the-rapids-jars
image
is this intentional to keep the sample version as snapshot here?

**Comment by sameerz** [comment]
> hi looks like this PR did not include version update to https://github.com/NVIDIA/spark-rapids/blob/branch-21.08/docs/get-started/getting-started-on-prem.md#download-the-rapids-jars
> is this intentional to keep the sample version as snapshot here?

That was a miss.  I also updated the README.md for the integration_tests.

**Review by jlowe — APPROVED** [review]

**Review by jlowe — DISMISSED** [review]

**Comment by sameerz** [comment]
build

**Comment by sameerz** [comment]
@pxLi @GaryShen2008 @NvTimLiu this is ready to merge right before you create the CHANGELOG for 21.08.

**Comment by sameerz** [comment]
build

**Review by jlowe — APPROVED** [review]

**Comment by pxLi** [comment]
As cudf released [v21.08.02](https://github.com/rapidsai/cudf/releases/tag/v21.08.02), I think we could go directly w/ dep cudfjni 21.08.2,
`cudf-21.08.0-cuda11` -> `cudf-21.08.2-cuda11`

File: docs/configs.md [hunk 3]
```diff
@@ -10,7 +10,7 @@ The following is the list of options that `rapids-plugin-4-spark` supports.
 On startup use: `--conf [conf key]=[conf value]`. For example:
 
 ```
-${SPARK_HOME}/bin/spark --jars 'rapids-4-spark_2.12-21.08.0-SNAPSHOT.jar,cudf-21.08.0-SNAPSHOT-cuda11.jar' \
+${SPARK_HOME}/bin/spark --jars 'rapids-4-spark_2.12-21.08.0.jar,cudf-21.08.0-cuda11.jar' \
```
Comment (pxLi) [inline docs/configs.md:13]:
```suggestion
${SPARK_HOME}/bin/spark --jars 'rapids-4-spark_2.12-21.08.0.jar,cudf-21.08.2-cuda11.jar' \
```

File: docs/demo/Databricks/generate-init-script-cuda11.ipynb [hunk 4]
```diff
@@ -1 +1 @@
...
....maven.org/maven2/com/nvidia/rapids-4-spark_2.12/21.08.0/rapids-4-spark_2.12-21.08.0.jar\nsudo wget -O /databricks/jars/cudf-21.08.0-cuda11.jar https://repo1.maven.org/maven2/ai/rapids/cudf/21.08.0/cudf-21.08.0-cuda11.jar\n\nsudo wget -O /etc/apt/preferences.d/cuda-repository-pin-600 https://developer.download.nvidia.com/compute/cuda/repos/ubuntu1804/x86_64/cuda-ubuntu1804.pin\nsudo wget -O ~/cuda-repo-ubuntu1804-11-0-local_11.0.3-450.51.06-1_amd64.deb https://developer.download.nvidia.com/compute/cuda/11.0.3/local_installers/cuda-repo-ubuntu1804-11-0-local_11.0.3-450.51.06-1_amd64.deb\nsudo dpkg -i ~/cuda-repo-ubuntu1804-11-0-local_11.0.3-450.51.06-1_amd64.deb\nsudo apt-key add /var/cuda-repo-ubuntu1804-11-0-local/7fa2af80.pub\nsudo apt-get update\nsudo apt -y install cuda-toolkit-11-0\"\"\", True)"],"metadata":{},"outputs":[],"execution_count":1},{"cell_type":"code","source":["%sh\ncd ../../dbfs/databricks/init_scripts\npwd\nls -ltr\ncat init.sh"],"metadata":{},"outputs":[],"execution_count":2},{"cell_type":"code","source":[""],"metadata":{},"outputs":[],"execution_count":3}],"metadata":{"name":"generate-init-script","notebookId":2645746662301564},"nbformat":4,"nbformat_minor":0}
```
Comment (pxLi) [inline docs/demo/Databricks/generate-init-script-cuda11.ipynb:1]:
```suggestion
{"cells":[{"cell_type":"code","source":["dbutils.fs.mkdirs(\"dbfs:/databricks/init_scripts/\")\n \ndbutils.fs.put(\"/databricks/init_scripts/init.sh\",\"\"\"\n#!/bin/bash\nsudo wget -O /databricks/jars/rapids-4-spark_2.12-21.08.0.jar https://repo1.maven.org/maven2/com/nvidia/rapids-4-spark_2.12/21.08.0/rapids-4-spark_2.12-21.08.0.jar\nsudo wget -O /databricks/jars/cudf-21.08.2-cuda11.jar https://repo1.maven.org/maven2/ai/rapids/cudf/21.08.2/cudf-21.08.2-cuda11.jar\n\nsudo wget -O /etc/apt/preferences.d/cuda-repository-pin-600 https://developer.download.nvidia.com/compute/cuda/repos/ubuntu1804/x86_64/cuda-ubuntu1804.pin\nsudo wget -O ~/cuda-repo-ubuntu1804-11-0-local_11.0.3-450.51.06-1_amd64.deb https://developer.download.nvidia.com/compute/cuda/11.0.3/local_installers/cuda-repo-ubuntu1804-11-0-local_11.0.3-450.51.06-1_amd64.deb\nsudo dpkg -i ~/cuda-repo-ubuntu1804-11-0-local_11.0.3-450.51.06-1_amd64.deb\nsudo apt-key add /var/cuda-repo-ubuntu1804-11-0-local/7fa2af80.pub\nsudo apt-get update\nsudo apt -y install cuda-toolkit-11-0\"\"\", True)"],"metadata":{},"outputs":[],"execution_count":1},{"cell_type":"code","source":["%sh\ncd ../../dbfs/databricks/init_scripts\npwd\nls -ltr\ncat init.sh"],"metadata":{},"outputs":[],"execution_count":2},{"cell_type":"code","source":[""],"metadata":{},"outputs":[],"execution_count":3}],"metadata":{"name":"generate-init-script","notebookId":2645746662301564},"nbformat":4,"nbformat_minor":0}
```

File: docs/demo/Databricks/generate-init-script.ipynb [hunk 5]
```diff
@@ -1 +1 @@
...
+{"cells":[{"cell_type":"code","source":["dbutils.fs.mkdirs(\"dbfs:/databricks/init_scripts/\")\n \ndbutils.fs.put(\"/databricks/init_scripts/init.sh\",\"\"\"\n#!/bin/bash\nsudo wget -O /databricks/jars/rapids-4-spark_2.12-21.08.0.jar https://repo1.maven.org/maven2/com/nvidia/rapids-4-spark_2.12/21.08.0/rapids-4-spark_2.12-21.08.0.jar\nsudo wget -O /databricks/jars/cudf-21.08.0-cuda11.jar https://repo1.maven.org/maven2/ai/rapids/cudf/21.08.0/cudf-21.08.0-cuda11.jar\"\"\", True)"],"metadata":{},"outputs":[],"execution_count":1},{"cell_type":"code","source":["%sh\ncd ../../dbfs/databricks/init_scripts\npwd\nls -ltr\ncat init.sh"],"metadata":{},"outputs":[],"execution_count":2},{"cell_type":"code","source":[""],"metadata":{},"outputs":[],"execution_count":3}],"metadata":{"name":"generate-init-script","notebookId":2645746662301564},"nbformat":4,"nbformat_minor":0}
```
Comment (pxLi) [inline docs/demo/Databricks/generate-init-script.ipynb:1]:
```suggestion
{"cells":[{"cell_type":"code","source":["dbutils.fs.mkdirs(\"dbfs:/databricks/init_scripts/\")\n \ndbutils.fs.put(\"/databricks/init_scripts/init.sh\",\"\"\"\n#!/bin/bash\nsudo wget -O /databricks/jars/rapids-4-spark_2.12-21.08.0.jar https://repo1.maven.org/maven2/com/nvidia/rapids-4-spark_2.12/21.08.0/rapids-4-spark_2.12-21.08.0.jar\nsudo wget -O /databricks/jars/cudf-21.08.2-cuda11.jar https://repo1.maven.org/maven2/ai/rapids/cudf/21.08.2/cudf-21.08.2-cuda11.jar\"\"\", True)"],"metadata":{},"outputs":[],"execution_count":1},{"cell_type":"code","source":["%sh\ncd ../../dbfs/databricks/init_scripts\npwd\nls -ltr\ncat init.sh"],"metadata":{},"outputs":[],"execution_count":2},{"cell_type":"code","source":[""],"metadata":{},"outputs":[],"execution_count":3}],"metadata":{"name":"generate-init-script","notebookId":2645746662301564},"nbformat":4,"nbformat_minor":0}
```

File: docs/download.md [hunk 6]
```diff
@@ -18,6 +18,61 @@ cuDF jar, that is either preinstalled in the Spark classpath on all nodes or sub
...
+
+	Python 3.6+, Scala 2.12, Java 8
+
+*Some hardware may have a minimum driver version greater than v450.80.02+.  Check the GPU spec sheet
+for your hardware's minimum driver version.
+
+### Download v21.08.0
+* Download the [RAPIDS
+  Accelerator for Apache Spark 21.08.0 jar](https://repo1.maven.org/maven2/com/nvidia/rapids-4-spark_2.12/21.08.0/rapids-4-spark_2.12-21.08.0.jar)
+* Download the [RAPIDS cuDF 21.08.0 jar](https://repo1.maven.org/maven2/ai/rapids/cudf/21.08.0/cudf-21.08.0-cuda11.jar)
```
Comment (pxLi) [inline docs/download.md:46]:
```suggestion
* Download the [RAPIDS cuDF 21.08.2 jar](https://repo1.maven.org/maven2/ai/rapids/cudf/21.08.2/cudf-21.08.2-cuda11.jar)
```

File: docs/get-started/Dockerfile.cuda [hunk 7]
```diff
@@ -50,8 +50,8 @@ COPY spark-3.0.2-bin-hadoop3.2/examples /opt/spark/examples
 COPY spark-3.0.2-bin-hadoop3.2/kubernetes/tests /opt/spark/tests
 COPY spark-3.0.2-bin-hadoop3.2/data /opt/spark/data
 
-COPY cudf-21.08.0-SNAPSHOT-cuda11.jar /opt/sparkRapidsPlugin
-COPY rapids-4-spark_2.12-21.08.0-SNAPSHOT.jar /opt/sparkRapidsPlugin
+COPY cudf-21.08.0-cuda11.jar /opt/sparkRapidsPlugin
```
Comment (pxLi) [inline docs/get-started/Dockerfile.cuda:53]:
```suggestion
COPY cudf-21.08.2-cuda11.jar /opt/sparkRapidsPlugin
```

File: docs/get-started/getting-started-on-prem.md [hunk 8]
```diff
@@ -55,17 +55,17 @@ CUDA and will not run on other versions. The jars use a maven classifier to keep
 - CUDA 11.0/11.1/11.2 => classifier cuda11
 
 For example, here is a sample version of the jars and cudf with CUDA 11.0 support:
-- cudf-21.08.0-SNAPSHOT-cuda11.jar
-- rapids-4-spark_2.12-21.08.0-SNAPSHOT.jar
+- cudf-21.08.0-cuda11.jar
```
Comment (pxLi) [inline docs/get-started/getting-started-on-prem.md:58]:
```suggestion
- cudf-21.08.2-cuda11.jar
```

File: docs/get-started/getting-started-on-prem.md [hunk 9]
```diff
@@ -55,17 +55,17 @@ CUDA and will not run on other versions. The jars use a maven classifier to keep
...
 jar that your version of the accelerator depends on.
 
 
 For simplicity export the location to these jars. This example assumes the sample jars above have
 been placed in the `/opt/sparkRapidsPlugin` directory:
 ```shell 
 export SPARK_RAPIDS_DIR=/opt/sparkRapidsPlugin
-export SPARK_CUDF_JAR=${SPARK_RAPIDS_DIR}/cudf-21.08.0-SNAPSHOT-cuda11.jar
-export SPARK_RAPIDS_PLUGIN_JAR=${SPARK_RAPIDS_DIR}/rapids-4-spark_2.12-21.08.0-SNAPSHOT.jar
+export SPARK_CUDF_JAR=${SPARK_RAPIDS_DIR}/cudf-21.08.0-cuda11.jar
```
Comment (pxLi) [inline docs/get-started/getting-started-on-prem.md:67]:
```suggestion
export SPARK_CUDF_JAR=${SPARK_RAPIDS_DIR}/cudf-21.08.2-cuda11.jar
```

File: integration_tests/README.md [hunk 10]
```diff
@@ -157,7 +157,7 @@ If you just want to verify the SQL replacement is working you will need to add t
 example assumes CUDA 11.0 is being used.
 
 ```
-$SPARK_HOME/bin/spark-submit --jars "rapids-4-spark_2.12-21.08.0-SNAPSHOT.jar,rapids-4-spark-udf-examples_2.12-21.08.0-SNAPSHOT.jar,cudf-21.08.0-SNAPSHOT-cuda11.jar" ./runtests.py
+$SPARK_HOME/bin/spark-submit --jars "rapids-4-spark_2.12-21.08.0.jar,rapids-4-spark-udf-examples_2.12-21.08.0.jar,cudf-21.08.0-cuda11.jar" ./runtests.py
```
Comment (pxLi) [inline integration_tests/README.md:160]:
```suggestion
$SPARK_HOME/bin/spark-submit --jars "rapids-4-spark_2.12-21.08.0.jar,rapids-4-spark-udf-examples_2.12-21.08.0.jar,cudf-21.08.2-cuda11.jar" ./runtests.py
```

File: integration_tests/README.md [hunk 11]
```diff
@@ -256,7 +256,7 @@ To run cudf_udf tests, need following configuration changes:
 As an example, here is the `spark-submit` command with the cudf_udf parameter on CUDA 11.0:
 
 ```
-$SPARK_HOME/bin/spark-submit --jars "rapids-4-spark_2.12-21.08.0-SNAPSHOT.jar,rapids-4-spark-udf-examples_2.12-21.08.0-SNAPSHOT.jar,cudf-21.08.0-SNAPSHOT-cuda11.jar,rapids-4-spark-tests_2.12-21.08.0-SNAPSHOT.jar" --conf spark.rapids.memory.gpu.allocFraction=0.3 --conf spark.rapids.python.memory.gpu.allocFraction=0.3 --conf spark.rapids.python.concurrentPythonWorkers=2 --py-files "rapids-4-spark_2.12-21.08.0-SNAPSHOT.jar" --conf spark.executorEnv.PYTHONPATH="rapids-4-spark_2.12-21.08.0-SNAPSHOT.jar" ./runtests.py --cudf_udf
+$SPARK_HOME/bin/spark-submit --jars "rapids-4-spark_2.12-21.08.0.jar,rapids-4-spark-udf-examples_2.12-21.08.0.jar,cudf-21.08.0-cuda11.jar,rapids-4-spark-tests_2.12-21.08.0.jar" --conf spark.rapids.memory.gpu.allocFraction=0.3 --conf spark.rapids.python.memory.gpu.allocFraction=0.3 --conf spark.rapids.python.concurrentPythonWorkers=2 --py-files "rapids-4-spark_2.12-21.08.0.jar" --conf spark.executorEnv.PYTHONPATH="rapids-4-spark_2.12-21.08.0.jar" ./runtests.py --cudf_udf
```
Comment (pxLi) [inline integration_tests/README.md:259]:
```suggestion
$SPARK_HOME/bin/spark-submit --jars "rapids-4-spark_2.12-21.08.0.jar,rapids-4-spark-udf-examples_2.12-21.08.0.jar,cudf-21.08.2-cuda11.jar,rapids-4-spark-tests_2.12-21.08.0.jar" --conf spark.rapids.memory.gpu.allocFraction=0.3 --conf spark.rapids.python.memory.gpu.allocFraction=0.3 --conf spark.rapids.python.concurrentPythonWorkers=2 --py-files "rapids-4-spark_2.12-21.08.0.jar" --conf spark.executorEnv.PYTHONPATH="rapids-4-spark_2.12-21.08.0.jar" ./runtests.py --cudf_udf
```

File: sql-plugin/src/main/scala/com/nvidia/spark/rapids/RapidsConf.scala [hunk 12]
```diff
@@ -1280,7 +1280,7 @@ object RapidsConf {
         |On startup use: `--conf [conf key]=[conf value]`. For example:
         |
         |```
-        |${SPARK_HOME}/bin/spark --jars 'rapids-4-spark_2.12-21.08.0-SNAPSHOT.jar,cudf-21.08.0-SNAPSHOT-cuda11.jar' \
+        |${SPARK_HOME}/bin/spark --jars 'rapids-4-spark_2.12-21.08.0.jar,cudf-21.08.0-cuda11.jar' \
```
Comment (pxLi) [inline sql-plugin/src/main/scala/com/nvidia/spark/rapids/RapidsConf.scala:1283]:
```suggestion
        |${SPARK_HOME}/bin/spark --jars 'rapids-4-spark_2.12-21.08.0.jar,cudf-21.08.2-cuda11.jar' \
```

**Comment by pxLi** [comment]
We are going to deploy cudf jni 21.08.2 to maven central, I will merge this doc update first and will cover the cudf 21.08.0 -> 21.08.2 doc update in https://github.com/NVIDIA/spark-rapids/pull/3146

**Pull request merged**